File: 423.go
func originalDigits(s string) string {
	//ans := make(char[], 0)
	var ans string = ""
	hashmap := make(map[byte]int)
	for i := 0; i < len(s); i++ {
		hashmap[s[i]]++
	}
	num := make([]int, 10)
	num[0] = hashmap['z']
	num[2] = hashmap['w']
	num[6] = hashmap['x']
	num[8] = hashmap['g']
	//num[1] = hashmap['o'] - num[0] - num[2]
	num[3] = hashmap['t'] - num[2] - num[8]
	num[7] = hashmap['s'] - num[6]
	num[4] = hashmap['r'] - num[0] - num[3]
	num[5] = hashmap['f'] - num[4]
	num[1] = hashmap['o'] - num[0] - num[2] - num[4]
	num[9] = (hashmap['n'] - num[1] - num[7]) / 2

	for i := 0; i < 10; i++ {
		for num[i] > 0 {
			num[i]--
			ans += strconv.Itoa(i)
		}
	}
	return ans
}

File: 423.java
class Solution {
    public String originalDigits(String s) {
        StringBuffer ss = new StringBuffer();
        HashMap<Character, Integer> map = new HashMap<>();
        int[] num = new int[10];
        for(int i = 0;i < s.length();i ++) map.put(s.charAt(i), map.getOrDefault(s.charAt(i), 0)+1);
        num[0] = map.getOrDefault('z', 0);
        num[2] = map.getOrDefault('w', 0);
        num[6] = map.getOrDefault('x', 0);
        num[8] = map.getOrDefault('g', 0);
        num[3] = map.getOrDefault('t', 0) - num[2] - num[8];
        num[7] = map.getOrDefault('s', 0) - num[6];
        num[4] = map.getOrDefault('r', 0) - num[0] - num[3];
        num[1] = map.getOrDefault('o', 0) - num[0] - num[2] - num[4];
        num[5] = map.getOrDefault('f', 0) - num[4];
        num[9] = (map.getOrDefault('n', 0) - num[1] - num[7])/2;
        
        for(int i = 0;i < 10;i ++) {
            while(num[i] > 0) {
                num[i] --;
                ss.append(i + "");
            }
        }
        return ss.toString();
    }
}